scalafix/rules/src/main/scala/fix/v0_14_0/FixLogicalTypeSupplier.scala

```diff
@@ -0,0 +1,87 @@
+package fix.v0_14_0
+
+import scalafix.v1._
+import scala.meta._
+
+object FixLogicalTypeSupplier {
+  val ParquetConfigurationMatcher: SymbolMatcher = SymbolMatcher.normalized(
+    "com/spotify/scio/parquet/package/ParquetConfiguration#of"
+  )
+  val SetClassMatcher: SymbolMatcher = SymbolMatcher.normalized(
+    "org/apache/hadoop/conf/Configuration#setClass"
+  )
+
+  val JavaClassMatcher: SymbolMatcher = SymbolMatcher.normalized("java/lang/Class")
+
+  private val ParquetAvroPrefix = "com/spotify/scio/parquet/avro"
+  val LogicalTypeSupplierMatcher: SymbolMatcher = SymbolMatcher.normalized(
+    s"$ParquetAvroPrefix/LogicalTypeSupplier",
+    "org/apache/beam/sdk/extensions/smb/AvroLogicalTypeSupplier"
+  )
+
+  private val ParquetAvroMatcher = SymbolMatcher.normalized(
+    s"$ParquetAvroPrefix/syntax/ScioContextOps#parquetAvroFile",
+    s"$ParquetAvroPrefix/syntax/SCollectionOps#saveAsParquetAvroFile"
+  )
+}
+
+class FixLogicalTypeSupplier extends SemanticRule("FixLogicalTypeSupplier") {
+  import FixLogicalTypeSupplier._
+
+  private def isLogicalTypeSupplier(term: Term)(implicit doc: SemanticDocument): Boolean =
+    term match {
+      case q"classOf[$tpe]" => LogicalTypeSupplierMatcher.matches(tpe.symbol)
+      case _ =>
+        term.symbol.info
+          .map(_.signature)
+          .collect { case MethodSignature(_, _, returnedType) => returnedType }
+          .collect { case TypeRef(_, sym, tpe :: Nil) if JavaClassMatcher.matches(sym) => tpe }
+          .collect { case TypeRef(_, sym, _) => sym }
+          .exists(LogicalTypeSupplierMatcher.matches)
```

Comment:
This works when class is stored in a val.
Unfortunately for this shift, since the class is removed, the val needs to also go away but that's kind of hard to do.

Reply:
yeah, I didn't try to implement that due to the complexity